InnerEye/ML/run_ml.py

```diff
@@ -208,75 +273,117 @@ def set_run_tags_from_parent(self) -> None:
                                                run_context=RUN_CONTEXT)
         checkpoint_handler.download_recovery_checkpoints_or_weights()
+        trainer: Optional[Trainer] = None
         # do training and inference, unless the "only register" switch is set (which requires a run_recovery
         # to be valid).
         if not self.azure_config.only_register_model:
-            # Set local_dataset to the mounted path specified in azure_runner.py, if any, or download it if that fails
-            # and config.local_dataset was not already set.
-            self.model_config.local_dataset = self.mount_or_download_dataset()
-            # Check for existing dataset.csv file in the correct locations. Skip that if a dataset has already been
-            # loaded (typically only during tests)
-            if self.model_config.dataset_data_frame is None:
-                assert self.model_config.local_dataset is not None
-                ml_util.validate_dataset_paths(
-                    self.model_config.local_dataset,
-                    self.model_config.dataset_csv)
-
             # train a new model if required
             if self.azure_config.train:
                 with logging_section("Model training"):
-                    model_train(self.model_config, checkpoint_handler, num_nodes=self.azure_config.num_nodes)
-            else:
+                    trainer, _ = model_train(checkpoint_handler,
+                                             container=self.container,
+                                             num_nodes=self.azure_config.num_nodes)
+                # log the number of epochs used for model training
+                RUN_CONTEXT.log(name="Train epochs", value=self.container.num_epochs)
+            elif isinstance(self.container, InnerEyeContainer):
                 self.model_config.write_dataset_files()
                 self.create_activation_maps()
 
-            # log the number of epochs used for model training
-            RUN_CONTEXT.log(name="Train epochs", value=self.model_config.num_epochs)
-
-        # We specify the ModelProcessing as DEFAULT here even if the run_recovery points to an ensemble run, because
-        # the current run is a single one. See the documentation of ModelProcessing for more details.
-        self.run_inference_and_register_model(checkpoint_handler, ModelProcessing.DEFAULT)
-
-        if self.model_config.generate_report:
-            self.generate_report(ModelProcessing.DEFAULT)
-
-        # If this is an cross validation run, and the present run is child run 0, then wait for the sibling runs,
-        # build the ensemble model, and write a report for that.
-        if self.model_config.number_of_cross_validation_splits > 0:
-            if self.model_config.should_wait_for_other_cross_val_child_runs():
-                self.wait_for_runs_to_finish()
-                self.create_ensemble_model()
+        if isinstance(self.container, InnerEyeContainer):
```

Comment:
Shouldn't reports be generated for models specified as lightning containers as well?

Reply:
thanks, done.